### PR TITLE
[T4A3][T16-B2]ZhaoDingfan

### PR DESCRIPTION
**Pull request by dfz2019**

### CODE REVIEW

**Comment by dfz2019** [comment]
Ready for review.

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Contact.java:5]: Missing header comment. All non trivial methods should have javadoc format header comments.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private final String value;
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Contact.java:6]: Good job on making `value` private

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        contact = contact.trim();
+        this.value = contact;
+    }
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Contact.java:13]: `MESSAGE_ADDRESS_CONSTRAINTS` and `ADDRESS_VALIDATION_REGEX` could have been passed to the parent class' constructor such that `isValidAddress` can be inherited from the parent class as well. This also ensures that the contact detail is valid before assigning it to `value`.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -34,25 +32,4 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java:48]: `equals` should not be left to the parent class as objects from child classes need to be compared with their respective classes, and not with their parent class.

File: src/seedu/addressbook/data/person/Email.java [hunk 5]
```diff
@@ -58,4 +56,5 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+    */
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Email.java:59]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

**Comment by PierceAndy** [comment]
- Avoid accumulating unrelated changes into a single PR. 
  Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.
- The child classes should have made use of the parent class' constructor instead of duplicating code in both classes.

@ZhaoDingfan Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**